### DISCUSSION TITLE
Hide rows in table for which the current user do not have permission to view and edit

### DISCUSSION THREAD

**Discussion by on3ss** [filamentphp/filament#2682] [Help] [answered]
I noticed that filament already hides the edit button and disabled the click on the table for rows which the current user do have have permission to edit or view when a policy is set for that model. Instead I want to remove those rows from the table completely. Is there a way to do that with filament without manually applying scopes to the model based on current user role or permission?

**Comment by danharrin** [comment, marked as answer]
It's much better to add the scope to the Eloquent query, since looping through records to check for permissions is not performant.

**Reply by on3ss** [reply]
That makes sense. Awesome package btw. Thank you.

**Comment by lourbert** [comment]
If you need more control over how the rows are filtered, you can use a custom query closure. For example, you could use a closure to filter the rows based on the current user's role:
 
  public static function table(Table $table)
    {
        return $table
            ->columns([
                //..
            ])
            ->actions([
                Tables\Actions\ViewAction::make(),
                Tables\Actions\EditAction::make(),
            ])
            ->query(function (User $query) {
                return $query->where('role', auth()->user()->role);
            });
    }

**Reply by kevinrrm** [reply]
Hi. I tried your solution but I got this:
An attempt was made to evaluate a closure for [Filament\Tables\Table], but [$query] was unresolvable.

**Reply by xuanbach1992** [reply]
I think I should use modifyQueryUsing  replace query
